### PR TITLE
SCP-1888 Per-session funds 2

### PR DESCRIPTION
**Pull request by nau**
Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [x] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [x] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [x] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues
- If you changed any Purescript files:
    - [x] `nix-shell shell.nix --run fix-purty` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [x] Commits have useful messages
- [x] Review clarifications made it into the code
- [x] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: plutus-pab/src/Cardano/Wallet/Mock.hs [hunk 1]
```diff
@@ -91,11 +93,17 @@ handleMultiWallet = do
...
-            let pkh = pubKeyHash (toPublicKey privateKey)
+            let pubKey = toPublicKey privateKey
+            let pkh = pubKeyHash pubKey
             let walletId = byteString2Integer (getPubKeyHash pkh)
             let wallet = Wallet walletId
             let wallets' = Map.insert wallet privateKey wallets
             put wallets'
+            let walletState = WalletState privateKey2 emptyNodeClientState mempty (defaultSigningProcess (Wallet 2))
+
+            let action :: Eff '[WAPI.WalletEffect] Tx
```
Comment (j-mueller) [inline plutus-pab/src/Cardano/Wallet/Mock.hs:104]: Comment on what this does

Comment (j-mueller) [reply plutus-pab/src/Cardano/Wallet/Mock.hs:104]: Or maybe just pull it out to the top level and give it a good name

File: plutus-pab/src/Cardano/Wallet/Mock.hs [hunk 2]
```diff
@@ -91,11 +93,17 @@ handleMultiWallet = do
...
             let bytes = BS.pack . unpack $ seed
             let privateKey = PrivateKey (KB.fromBytes bytes)
-            let pkh = pubKeyHash (toPublicKey privateKey)
+            let pubKey = toPublicKey privateKey
+            let pkh = pubKeyHash pubKey
             let walletId = byteString2Integer (getPubKeyHash pkh)
             let wallet = Wallet walletId
             let wallets' = Map.insert wallet privateKey wallets
             put wallets'
+            let walletState = WalletState privateKey2 emptyNodeClientState mempty (defaultSigningProcess (Wallet 2))
```
Comment (j-mueller) [inline plutus-pab/src/Cardano/Wallet/Mock.hs:102]: why `privateKey2`? And have you tested this? I'm not sure wallet 2 will know which funds to use when it runs the action later on.

**Review by j-mueller — COMMENTED** [review]
Have you tested this?

File: plutus-pab/src/Cardano/Wallet/Mock.hs (hunk 2 above)
Comment (nau) [reply plutus-pab/src/Cardano/Wallet/Mock.hs:102]: That's a weird one. It didn't work with Wallet 1/privKey1. But it works with privKey2, I tested it.
Don't know why, but I want to move to companion contract asap.

Comment (nau) [reply plutus-pab/src/Cardano/Wallet/Mock.hs:102]: I tried to create multiple wallets, and it worked semi-fine. It create 2 or 3 successfully, and then the REST API call just hanged.

**Review by j-mueller — APPROVED** [review]

**Pull request merged**